### PR TITLE
Added zZenossGateway setting on zenpack install based on route info

### PR DESCRIPTION
**Pull request by ssvsergeyev**

### CODE REVIEW

File: ZenPacks/zenoss/Layer2/__init__.py [hunk 1]
```diff
@@ -38,8 +41,25 @@ class ZenPack(ZenPackBase):
...
+        self._getDefaultGateway(app.zport.dmd)
 
     def _buildDeviceRelations(self):
         # TODO: figure out how this is usefull and remove if it is not.
         for d in self.dmd.Devices.getSubDevicesGen():
             d.buildRelations()
+
+    def _getDefaultGateway(self, dmd):
+        """
+        Try to determine zZenossGateway value from /proc/net/route information
```
Comment (ssbunyk) [inline ZenPacks/zenoss/Layer2/__init__.py]: Why not from `route`? 

```
$ cat /proc/net/route 
Iface	Destination	Gateway 	Flags	RefCnt	Use	Metric	Mask		MTU	Window	IRTT                                                       
eth3	0096A8C0	00000000	0001	0	0	1	00FFFFFF	0	0	0                                                                               
eth3	00000000	0296A8C0	0003	0	0	0	00000000	0	0	0                                                                               
$ route -n
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
192.168.150.0   0.0.0.0         255.255.255.0   U     1      0        0 eth3
0.0.0.0         192.168.150.2   0.0.0.0         UG    0      0        0 eth3
```

It looks more readable and more easy to parse for me.

File: ZenPacks/zenoss/Layer2/__init__.py [hunk 2]
```diff
@@ -38,8 +41,25 @@ class ZenPack(ZenPackBase):
...
+        """
+
+        with open("/proc/net/route") as fh:
+            for line in fh:
+                fields = line.strip().split()
+                if fields[1] != '00000000' or not int(fields[3], 16) & 2:
```
Comment (ssbunyk) [inline ZenPacks/zenoss/Layer2/__init__.py]: Condition not very obvious and shoud be commented, espacially second part.

File: ZenPacks/zenoss/Layer2/__init__.py [hunk 3]
```diff
@@ -38,8 +41,25 @@ class ZenPack(ZenPackBase):
...
+                    continue
+
+                val = socket.inet_ntoa(struct.pack("<L", int(fields[2], 16)))
```
Comment (ssbunyk) [inline ZenPacks/zenoss/Layer2/__init__.py]: This transformation also should be commented.

File: ZenPacks/zenoss/Layer2/__init__.py (hunk 1 above)
Comment (byshovets) [inline ZenPacks/zenoss/Layer2/__init__.py]: /proc/net/route should be available in each Linux-based distro and /sbin/route is an external utility.

Comment (ssvsergeyev) [inline ZenPacks/zenoss/Layer2/__init__.py]: Vadym has a point here

**Pull request merged**